samcli/local/docker/lambda_container.py

```diff
@@ -201,3 +217,29 @@ def _get_entry_point(runtime, debug_options=None):  # pylint: disable=too-many-b
             runtime=runtime,
             options=LambdaContainer._DEBUG_ENTRYPOINT_OPTIONS,
         )
+
+    @staticmethod
+    def _get_additional_volumes(host_volumes):
+        """
+        Generate a dictionary for mounting additional volumes in container.
+        A remote directory is generated based on remote base path + host node folder name
+
+        :param Tuple(Path) host_volumes: Volumes to be mounted in a container.
+        :return dict: A dictionary containing volumes mapping between host and remote container
+        """
+
+        if not host_volumes:
+            return None
+
+        volume_map = {}
+        for host_volume in host_volumes:
+            host_base_names = os.path.basename(host_volume)
+
+            if not host_base_names:
+                continue
+
+            remote_volume_path = os.path.join(LambdaContainer._VOLUME_MOUNT_PATH, host_base_names)
+
+            volume_map[host_volume] = {"bind": remote_volume_path, "mode": ""}
```

Comment:
We have historically taken the stance that we won't do "write" mounts. Primarily for security, but it also helps performance.

What is your use case for writable mounts?

Reply:
We use a custom debugger application in Rider to debug Lambda in container. This debugger app file some log files that I would like to collect to investigate in cases of any issues. We configure our debugger with a custom log folder that I would like to map inside a container. We need write access to be able to write logs into a folder.

Reply:
Understood. Writable folders can potentially cause security attacks. We have to do a threat model and security posture assessment before we can enable this. If this is exclusively for debugging during development, I can help you find a workaround to `docker cp` logs.

Reply:
Would this work for youi?

Reply:
@sanathkr, sorry for such a long response. The main issue with `docker cp` approach is that we do not have a handler that allows us to copy logs before container is deleted. SAM CLI runs with `--rm` flag and we have no guarantee that container is alive at the moment we do `docker cp`. As a workaround, we can make this API on Rider side to make a sync call before we kill the process we want our logs from. I just would like to understand if this is the only option here. I totally understand the security issue with suggested approach. Please let me know if you think it is the stopper for this PR. If yes, please feel free to close and we will do a workaround on our side.